### ISSUE TITLE
Foreign Key Relationship in data model?

### ISSUE THREAD

**Issue by realdah** [typicode/json-server#193]
As per instruction, if we have below structure:

```
{
  "posts": [
    { "id": 1, "title": "json-server", "author": "typicode" }
  ],
  "comments": [
    { "id": 1, "body": "some comment", "postId": 1 }
  ],
  "profile": { "name": "typicode" }
}
```

Because we have "postId" in **comments**, so it is just like a foreign key from **comments** to **posts**.

Now, if I try

GET http://localhost:3000/posts/1/comments
We will get all **comments** with "postId" equals to **1** 

Until now, everything works just fine by default, no special route required.

However, the problem comes when I start to try:
POST http://localhost:3000/posts/1/comments with some basic data, and it says **404** error!!!
````
{
    key: "someData"
}
````

I know that I can still directly post to http://localhost:3000/comments with below data
````
{
    postId : 1,
    key : "someData"
}
````

But this is not consistent with the first **GET**.... 

I tried to find some more documents, but very less information. please help!

**Comment by marcelojeff** [comment]
I'm having the same issue. 

The route works with **GET**, but don't with **POST**

**Comment by marcelojeff** [comment]
The issue happens when trying to get some specific resource too. As @realdah said:

If I try **GET** `http://localhost:3000/posts/1/comments` returns me a list

But if I try **GET** `http://localhost:3000/posts/1/comments/1` returns me a **404**

**Comment by typicode** [comment]
@realdah you're right it makes more sense this way, I've added `POST` for nested resources in `v0.8.3`. Thanks for the suggestion.

**Comment by urecio** [comment]
@typicode, are you supporting GET and DELETE on nested resources. I cannot figure out how to deal with those in example. For example, after the creation of the nested comment, GET `/posts/1/comments/1` and DELETE will return 404. I could find a workaround with a route ending with `comments?postId=:postId` but still it isn't possible to DELETE

**Comment by typicode** [comment]
Hi @urecio, to delete, use `DELETE /comments/1`. The idea is that only comment id is useful to delete a comment, so there's no need to have a post id in the route.

**Comment by urecio** [comment]
That's true @typicode. In fact, I'm working on a different example, which is a many-to-many relationship. For example with the typical student-class relation, my current workaround is a bit tricky: 
`POST /classes`
`POST /students`
`POST /classes/1/students` with data => `{studentId: 1}`
- To get the students of a class I have to `GET /classes/1?_embed=students` and then iterate over the results...
- to delete I'm using the id autogenerated on students. A set looks like: {id: 2, studentId: 1, classId: 1}, so `DELETE /students/2`

Do you think I'm misunderstanding something and overcomplicating it?
